Add a Snackbar notice to inform users that settings have been saved successfully.

iamdharmesh opened the pull request:
### Description of the Change
This PR adds a Snackbar notice to inform users that settings have been saved successfully.

![Dec-23-2024 22-31-06](https://github.com/user-attachments/assets/46a8c1cd-5315-4511-8a1e-9fedb1e2f718)


Closes #837 

### How to test the Change
1. Navigate to the ClassifAI settings page (Tools > ClassifAI).
1. Enable or disable any feature and confirm that the notice is displayed properly.
1. Navigate to the settings page of any feature and save the settings.
1. Validate that the Snackbar notice confirming the settings have been saved is displayed.

### Changelog Entry
> Added - Snackbar notice to inform users that settings have been saved successfully.

### Credits
Props @iamdharmesh @dkotter @jeffpaul 

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [x] All new and existing tests pass.

iamdharmesh committed:
Add save settings snackbar notifications to inform users settings saved successfully.

iamdharmesh committed:
Add snackbar notice in classifai registration settings as well.

dkotter reviewed (commented):
Overall this works nice and code looks good here. Couple things worth discussing:

1. Is it possible to reduce the time the notice shows? Right now the snackbar sticks around for quite a while and I think it would be nice if we could hide it quicker
2. This solves success messages but doesn't solve error messages. If you save settings for a Feature and an error occurs, the error shows up top but it is often out of view. Would be great if we scroll to the top when that happens so the error can be seen

jeffpaul commented:
Note that there's discussion in WP/GB via https://github.com/WordPress/gutenberg/issues/16391 on overall accessibility of the snackbar, so we might perhaps review the consensus from a11y folks there and ensure we utilize the snackbar for error & success states in a solid a11y manner.

jeffpaul commented:
Looks like https://github.com/WordPress/gutenberg/issues/67662 is an updated proposal in WP/GB that we might consider here as well.

iamdharmesh committed:
Add scroll to error notice.

iamdharmesh commented:
@dkotter I have added a scroll-to-top to make error notices more visible. Regarding the Snackbar notice, as @jeffpaul pointed out, it has open accessibility issues. Since we have already added the scroll-to-top, I think we can replace it with a standard success notice combined with the scroll-to-top. However, I am not too concerned about this since the message we are displaying is not high priority. Additionally, the core itself uses Snackbar notices, and I have seen some plugins, such as ElasticPress, doing the same. what do you think?

dkotter commented:
>  Additionally, the core itself uses Snackbar notices, and I have seen some plugins, such as ElasticPress, doing the same. what do you think?

I'm fine with using the Snackbar here. We can always change this once there's a better option in Core.

> I have added a scroll-to-top to make error notices more visible.

Thanks, this works great.

One last thing is how long the snackbar shows. If we can't reduce the time that shows (not sure if that's a setting that is supported by Core) I think we should at least remove the snackbar (if this is easy enough) anytime the screen changes.

As an example, if I enable a Feature, the snackbar shows. If I then click in to configure the settings for that Feature, ideally we remove the snackbar at that point.

Then when settings are saved, we show the snackbar again. If I go back to the main screen, that snackbar should be removed again.

iamdharmesh committed:
Remove snackbar location change.

iamdharmesh commented:
>One last thing is how long the snackbar shows. If we can't reduce the time that shows (not sure if that's a setting that is supported by Core) I think we should at least remove the snackbar (if this is easy enough) anytime the screen changes.

It seems the time limit is [hardcoded](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/snackbar/index.tsx#L29). We may be able to reduce the time by setting a timeout function on our side, but I think removing the snackbar on screen change looks more appropriate here. I have updated the code to remove notices on location change. Could you please help review it once? Thanks.
